spark/src/main/scala/org/apache/iceberg/spark/SparkTableUtil.scala

```diff
@@ -297,5 +301,88 @@ object SparkTableUtil {
       )
     }
   }
+
+  def buildManifest(table: Table,
+                    sparkDataFiles: Seq[SparkDataFile],
+                    partitionSpec: PartitionSpec): ManifestFile = {
+    val outputFile = table.io
+      .newOutputFile(FileFormat.AVRO.addExtension("/tmp/" + UUID.randomUUID.toString))
+    val writer = ManifestWriter.write(partitionSpec, outputFile)
+    try {
+      for (file <- sparkDataFiles) {
+        writer.add(file.toDataFile(partitionSpec))
+      }
+    } finally {
+      writer.close()
+    }
+
+    writer.toManifestFile
+  }
+
+  def partitionToMap(partition: String): Map[String, String] = {
+    val map = new mutable.HashMap[String, String]()
+    val list = partition.split("/")
+    list.foreach { str =>
+      val kv = str.split("=")
+      map.put(kv(0), kv(1))
+    }
+
+    map.toMap
+  }
+
+  /**
+   * Migrate a spark table to a iceberg table.
+   *
+   * The migration uses the spark session to get table metadata. It assumes no
+   * operation is going on original table and target table and thus is not
+   * thread-safe.
+   *
+   * @param dbName the database name of the table to be migrated
+   * @param tableName the table to be migrated
+   * @param table the target table to migrate in
+   *
+   * @return table the target table
+   */
+  def migrateSparkTable(dbName: String, tableName: String, table: Table): Table = {
```

Comment:
The logic in this method sounds more like "import data" rather than "migrate" to me. Migration should probably replace a pointer in the catalog.

Reply:
Aha, it does like the "import data" behaviour. When I tried to use the iceberg, the first thing I had to do is to "import table".  If we want to implement a migration include replace the pointer, more steps should be needed. I can have a look further.  At this stage, I 'd like to implement "import data" since that should also be needed at first in the migration IIUC.

Reply:
I totally agree. I think we cannot really implement "migrate" before we have proper DS V2 APIs coming in Spark 3.0. My comment was only about naming.